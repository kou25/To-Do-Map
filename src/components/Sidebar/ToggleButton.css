.button{
    display: flex;
    flex-direction: column;
    justify-content: space-around;
    height: 30px;
    width: 36px;
    background: transparent;
    border: none;
    cursor: pointer;
    padding: 0;
    box-sizing: border-box;
}

.button:focus{
    outline: none;
}

.line{
    height: 2px;
    width: 30px;
    background: white;

}

