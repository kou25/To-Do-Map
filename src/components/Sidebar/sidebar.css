.sidebar{
    height: 100%;
    background-color: white;
    box-shadow: 2px 0px 7px rgba(0,0,0,0.5);
    position: fixed;
    top: 0;
    left: 0;
    width: 40%;
    max-width: 400px;
    z-index: 200;
    transform: translateX(-100%);
    transition: transform 0.3s ease-out;
}
.sidebar.open{
    transform: translateX(0);
}

.sidebar ul{
    height: 100%;
    list-style: none;
    display: flex;
    flex-direction: column;
    margin-top: 50px;
}
i{
    padding-right: 0.5rem;
    padding-left: 0.5rem;
    color: chocolate;
}

.sidebar li{
    margin: 0.5rem 0;
    padding-bottom: 10px;
    margin-right: 3rem;
}

.sidebar a{
    color: black;
    text-decoration: none;
    font-size: 1.2rem;
    font-weight: 500;
    font-family: cursive;
}

.sidebar a:hover, .sidebar a:active{
    color: chocolate;
    font-size: 1.5rem;
    transition: font-size 0.5s ease-in;
}

.pro a{
   margin: 20px;
   font-size: 1.5rem;
   text-decoration: none;
   font-family: cursive;
   
}
hr{
    -webkit-box-shadow: 2px 2px 4px 0px rgba(0,0,0,0.75);
-moz-box-shadow: 2px 2px 4px 0px rgba(0,0,0,0.75);
box-shadow: 2px 2px 4px 0px rgba(0,0,0,0.75);
}

.sidebar li:nth-of-type(odd) {
    background-color: #e9e9f9 ;
}