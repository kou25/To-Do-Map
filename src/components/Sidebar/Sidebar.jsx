import React from 'react'
import './sidebar.css'
import profile from './profile.png'
import { Link } from 'react-router-dom'

const Sidebar =props=> {
    let barClasses='sidebar';
    if(props.show){
        barClasses ='sidebar open'
    }
    return (
            <nav className={barClasses}>
                <div className="pro">
                <Link  to='/signin' >
                    <img src={profile} alt="logo" style={{height:"70px"}}/>
                    <span style={{marginLeft:"40px"}}>Welcome</span>
                </Link>
                </div>
                <hr style={{backgroundColor:"orange", height:"1.2px", margin:"3px 5px"}}/>
                <ul>
                    <li data-testid="list"><i className="fa fa-line-chart" aria-hidden="true"></i><Link to='/dashboard'>Dashboard</Link><i className="fa fa-arrow-circle-right" aria-hidden="true" style={{marginLeft:"120px"}}></i></li>
                    <li data-testid="list1"><i className="fa fa-plus-circle" aria-hidden="true"></i><Link to='/add'>Add To-DO</Link><i className="fa fa-arrow-circle-right" aria-hidden="true" style={{marginLeft:"110px"}}></i></li>
                    <li data-testid="list2"><i className="fa fa-eye" aria-hidden="true"></i><Link to='/view'>View To-DO</Link><i className="fa fa-arrow-circle-right" aria-hidden="true" style={{marginLeft:"103px"}}></i></li>
                    <li data-testid="list3"><i className="fa fa-map-marker" aria-hidden="true"></i><Link to='/map'>View Map</Link><i className="fa fa-arrow-circle-right" aria-hidden="true"style={{marginLeft:"132px"}}></i></li>
                </ul>
            </nav>
    )
}

export default Sidebar
