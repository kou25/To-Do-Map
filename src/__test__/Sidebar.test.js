import React from 'react';
import ReactDom from 'react-dom'
import Sidebar from '../components/Sidebar/Sidebar';
import { BrowserRouter } from 'react-router-dom';
import {render, cleanup} from '@testing-library/react'
import "@testing-library/jest-dom/extend-expect"
import renderer from 'react-test-renderer'

afterEach(cleanup);
it("render nav without crashing",()=>{
    const nav= document.createElement("nav");
    ReactDom.render(
    <BrowserRouter>
    <Sidebar/>
    </BrowserRouter>,nav);
  })

it("render without crashing",()=>{
  const div= document.createElement("div");
  ReactDom.render(
  <BrowserRouter>
  <Sidebar/>
  </BrowserRouter>,div);
})

it("renders links correctly", ()=>{
    const{getByTestId}=render(
        <BrowserRouter>
        <Sidebar/>
        </BrowserRouter>)
    expect(getByTestId('list')).toHaveTextContent('Dashboard')
  })
  it("renders links correctly", ()=>{
    const{getByTestId}=render(
        <BrowserRouter>
        <Sidebar/>
        </BrowserRouter>)
    expect(getByTestId('list1')).toHaveTextContent('Add To-DO')
  })
  it("renders links correctly", ()=>{
    const{getByTestId}=render(
        <BrowserRouter>
        <Sidebar/>
        </BrowserRouter>)
    expect(getByTestId('list2')).toHaveTextContent('View To-DO')
  })
  it("renders links correctly", ()=>{
    const{getByTestId}=render(
        <BrowserRouter>
        <Sidebar/>
        </BrowserRouter>)
    expect(getByTestId('list3')).toHaveTextContent('View Map')
  })
  
  it('matches snapshot', ()=>{
    const tree=renderer.create(
        <BrowserRouter>
        <Sidebar/>
        </BrowserRouter>).toJSON()
    expect(tree).toMatchSnapshot()
  })

  

