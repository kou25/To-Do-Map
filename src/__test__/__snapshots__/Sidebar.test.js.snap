// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<nav
  className="sidebar"
>
  <div
    className="pro"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <img
        alt="logo"
        src="profile.png"
        style={
          Object {
            "height": "70px",
          }
        }
      />
      <span
        style={
          Object {
            "marginLeft": "40px",
          }
        }
      >
        Welcome
      </span>
    </a>
  </div>
  <hr
    style={
      Object {
        "backgroundColor": "orange",
        "height": "1.2px",
        "margin": "3px 5px",
      }
    }
  />
  <ul>
    <li
      data-testid="list"
    >
      <i
        aria-hidden="true"
        className="fa fa-line-chart"
      />
      <a
        href="/dashboard"
        onClick={[Function]}
      >
        Dashboard
      </a>
      <i
        aria-hidden="true"
        className="fa fa-arrow-circle-right"
        style={
          Object {
            "marginLeft": "120px",
          }
        }
      />
    </li>
    <li
      data-testid="list1"
    >
      <i
        aria-hidden="true"
        className="fa fa-plus-circle"
      />
      <a
        href="/add"
        onClick={[Function]}
      >
        Add To-DO
      </a>
      <i
        aria-hidden="true"
        className="fa fa-arrow-circle-right"
        style={
          Object {
            "marginLeft": "110px",
          }
        }
      />
    </li>
    <li
      data-testid="list2"
    >
      <i
        aria-hidden="true"
        className="fa fa-eye"
      />
      <a
        href="/view"
        onClick={[Function]}
      >
        View To-DO
      </a>
      <i
        aria-hidden="true"
        className="fa fa-arrow-circle-right"
        style={
          Object {
            "marginLeft": "103px",
          }
        }
      />
    </li>
    <li
      data-testid="list3"
    >
      <i
        aria-hidden="true"
        className="fa fa-map-marker"
      />
      <a
        href="/map"
        onClick={[Function]}
      >
        View Map
      </a>
      <i
        aria-hidden="true"
        className="fa fa-arrow-circle-right"
        style={
          Object {
            "marginLeft": "132px",
          }
        }
      />
    </li>
  </ul>
</nav>
`;
